glTF / MaterialX Translator Support

kwokcb opened the pull request:
## glTF / MaterialX bi-directional direct translation utilities

* MaterialXglTF / PyMaterialXglTF : New modules for translation utils. 
* gltf_coloredImage, gltf_image, gltf_normalmap, and gltf_iridescence already in ASWF main and used for import, and partially used for export.
* CgltfMaterialLoader : is the new basic reader / writer to handle the "target" graph. Anything else will not be understood
  and must be handled via graph translation or baking.
* Shader translation graph: Currently has WIP version from Autodesk to convert std_surface to gltf_pbr. To sync when core PR in.
* glTF-Sample-Models : If found unit tests will perform conversion 1:1 between non-embedded glTF files to mtlx material files.
These files are picked up as part of shader generation and render tests for Document, code, and render validation.

kwokcb committed:
Migrate private prototype code to Khronos branch.

kwokcb committed:
Work path names.

kwokcb committed:
Restore unique path check.

kwokcb committed:
Merge branch 'rework_pathing' into gltf_materialx_interop

kwokcb committed:
Turn off debug.

kwokcb committed:
Cleanup pathing.

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into gltf_materialx_interop

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/MaterialX?pullRequest=4) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/MaterialX?pullRequest=4) before we can accept your contribution.<br/>**0** out of **4** committers have signed the CLA.<br/><br/>:x: kwokcb<br/>:x: jstone-lucasfilm<br/>:x: niklasharrysson<br/>:x: crydalch<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/MaterialX?pullRequest=4) it.</sub>

kwokcb committed:
When building paths ensure that path created is converted to a valid name for both the geometry and material loader.
To log an issue that material assignment geometry string matching uses a per character match vs a string match so something like "/ foo" will match anything
with a "/" in the path.

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into gltf_materialx_interop

kwokcb committed:
Fix pathing in default shader ball.
Fix path naming to remove front "/" and only add mesh name if the parent transform path is empty.

kwokcb committed:
Patch last fix.

kwokcb committed:
Fix missing geometry.

kwokcb committed:
Merge branch 'main' of https://github.com/AcademySoftwareFoundation/MaterialX into gltf_materialx_interop

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into gltf_materialx_interop

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into gltf_materialx_interop

kwokcb committed:
Merge branch 'main' into gltf_materialx_interop

kwokcb committed:
Merge branch 'gltf_materialx_interop' of https://github.com/KhronosGroup/MaterialX into gltf_materialx_interop

kwokcb committed:
Create MaterialXglTF optional module. Builds by default in this fork.

kwokcb committed:
Fix export symbols.

kwokcb committed:
some normalmap + alpha updates.

kwokcb committed:
Turn off glTF module for shared builds due to linkage issues.

kwokcb committed:
Fix check-in.

kwokcb committed:
Migrate over gltf_coloredimage + test.

kwokcb committed:
Add support for unlit for import

kwokcb committed:
Misc channel updates

- alpha mapping from image
- remove extract node usage and use channels instead for
orm and basecolor/alpha
- misc cleanup.

kwokcb committed:
Fix js build.

kwokcb committed:
Fix unused var.

kwokcb committed:
Add in basic conversion to mtlx of glTF sample models. Once there they can go through genshader and render tests.
Already existing issue of __ as leading names for which is invalid in glsl has been logged in core.

kwokcb committed:
Handle more illegal identifier names.

kwokcb committed:
First rts run using glTF sample models. Filtering is required:
- Files have duplicate shaders not worth testing.
- Files have hundres of materials. Good for stress testing usage but not conversion.
- Some files use image types not loadable by testrender. Not a failure but should report this to OSL folks.
- Some files don't really test materials but other features and are essentially duplicaes or not worth testing (e.g. animation testing)

kwokcb committed:
Update gltf_colorimage definition. Use multiply instead of mix and remove swizzles for simplification.
Use proposed naming convention for 1.39 separate4 nodedef (not added yet).

kwokcb committed:
Fix test.

kwokcb committed:
Use gltf_coloredImage to handle color/alpha blending properly.

kwokcb committed:
Remove usage of "channels" as this will be deprecated in 1.39.

kwokcb committed:
Handle
- export of gltf_coloredImage
- import and export of separate occlusion from orm

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into gltf_materialx_interop

kwokcb committed:
Branch update.
- New defs
- Python bindings
- Texture transforms, cpv, uv set index

kwokcb committed:
Merge branch 'main' of https://github.com/AcademySoftwareFoundation/MaterialX into gltf_materialx_interop

kwokcb committed:
Merge branch 'main' of https://github.com/AcademySoftwareFoundation/MaterialX into gltf_materialx_interop

jstone-lucasfilm committed:
Subclass GeomPropDef from TypedElement

This changelist updates the parent class of GeomPropDef from Element to TypedElement, providing it with access to the setType and getType methods.

jstone-lucasfilm committed:
Improvements to node helper methods

- Fix an edge case in Node::setConnectedNode, omitting a type copy when the upstream node is multi-output.
- Add helper method GraphElement::addGeomNode, allowing geometry nodes to be constructed from GeomPropDef elements.

jstone-lucasfilm committed:
Improvements to findRenderableElements

- Only search for graph outputs when no material nodes are present.
- Simplify search logic for graph outputs.

jstone-lucasfilm committed:
Improvements to flattenSubgraphs (#1101)

- Extend flattenSubgraphs to handle a broader range of graph structures, including references to interfaces with 'defaultgeomprop' strings.
- Streamline the logic in flattenSubgraphs, merging duplicate code paths and clarifying documentation.
- Add a 'Flatten Subgraphs' option to the MaterialX viewer, allowing render comparisons between nested and flattened graphs.

jstone-lucasfilm committed:
Remove duplicate bindings in documents

This changelist removes duplicate data bindings from the library and example documents in MaterialX, allowing this case to be potentially flagged as a validation warning in a future update.

Additionally, a handful of unit tests that depended upon duplicate bindings being present have been removed from the test suite.

crydalch committed:
Adds a Unified Noise node (#1087)

Adds a unified noise node, for artists to have a single noise node that unifies shared controls (such as frequency and offset). A common complaint we hear from artists, is the MaterialX noises are too bare, they are missing the very basics of frequency/offset, and even some post-processing, which requires creating all sorts of nodes. Rather than modify all of the existing noises, it seemed a slightly higher-level node to encompass shared features made more sense (and would be easier to adopt).

Jitter is treated as a common control; it behaves like jitter does on the standard worley noise, and for the non-Worley noises it scrambles the input. The fractal controls are unique to that particular noise type, so an artist tool would probably hide those parameters when the noise type is not set to fractal.

crydalch committed:
Adds a Color Correct node (#1086)

Wraps up existing adjustment nodes into an artist-friendly color correction node.

jstone-lucasfilm committed:
Update versions in GitHub Actions

This changelist updates versions for several steps in the main GitHub Actions workflow, and upgrades the active version of Node.js from 12 to 16.

niklasharrysson committed:
Codegen improvements to surfaceshader mix operator (#1104)

This change list improves the handling of surfaceshader mix operator in code generation. One notable change is for OSL shader generation, which now implements the surfaceshader type as a struct of separate BSDF, EDF and Opacity components, in order to simplify operations like mixing.

Also adds new logic to HW transparency detection to support generations of GLSL shaders with transparency when mixing surface shaders.

kwokcb committed:
Fix parent / child assignment  logic.

kwokcb committed:
Review update

kwokcb committed:
Add "contrast" to disallowed list as it exists in mx_funcs.h and colorcorrect has a "contrast" arg so will conflict. (#1107)

kwokcb committed:
Merge branch 'main' of https://github.com/AcademySoftwareFoundation/MaterialX into gltf_materialx_interop

kwokcb committed:
Merge branch 'AcademySoftwareFoundation:main' into parent_child_assign

kwokcb committed:
Merge branch 'parent_child_assign' of https://github.com/kwokcb/MaterialX into gltf_materialx_interop
